lapce-data/src/config.rs

```diff
@@ -538,30 +538,91 @@ impl Config {
         self.themes.style_color(name)
     }
 
+    /// Calculate the width of the character "W" (being the widest character)
```

Comment:
I almost commented on the "W" argument before I saw this.

Since much of the editor rendering relies on fonts being monospaced, is "W" really the widest character? Relying on it certainly prevents us from supporting variable width fonts...

Reply:
I think it generally is, but it all depends on the font metrics of course. And yes, this entire functionality will have to be replaced with something else to support variable width fonts, which funnily enough is what kicked off my interest in contributing to Lapce. Another one is we need separate font settings for the editor and the terminal.